client/src/send_and_confirm_transactions_in_parallel.rs

```diff
@@ -0,0 +1,465 @@
+                        let statuses = result.value;
+                        for (signature, status) in signatures.iter().zip(statuses.into_iter()) {
+                            if let Some(status) = status {
+                                if status.satisfies_commitment(rpc_client.commitment()) {
+                                    if let Some((_, data)) = transaction_map.remove(signature) {
+                                        confirmed_transactions.fetch_add(1, Ordering::Relaxed);
+                                        if let Some(error) = status.err {
+                                            errors_map.insert(data.index, error);
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+
+                last_block_height = current_blockheight;
+            }
+            tokio::time::sleep(Duration::from_millis(100)).await;
+        }
+    })
+}
+
+#[allow(clippy::too_many_arguments)]
+async fn sign_all_messages_and_send<T: Signers + ?Sized>(
+    progress_bar: &Option<indicatif::ProgressBar>,
+    rpc_client: Arc<RpcClient>,
+    tpu_client: &Option<TpuClient<QuicPool, QuicConnectionManager, QuicConfig>>,
+    transaction_map: Arc<DashMap<Signature, TransactionData>>,
+    error_map: Arc<DashMap<usize, TransactionError>>,
+    messages_with_index: Vec<(usize, Message)>,
+    blockhash_data_rw: Arc<RwLock<BlockHashData>>,
+    signers: &T,
+    confirmed_transactions: Arc<AtomicU32>,
+    total_transactions: usize,
+) -> Result<()> {
+    let current_transaction_count = messages_with_index.len();
+    // send all the transaction meesages
+    for (counter, (index, message)) in messages_with_index.iter().enumerate() {
+        let mut transaction = Transaction::new_unsigned(message.clone());
+        let blockhashdata = *blockhash_data_rw.read().await;
+
+        // we have already checked if all transactions are signable.
+        transaction
+            .try_sign(signers, blockhashdata.blockhash)
+            .expect("Transaction should be signable");
+        let signature = transaction.signatures[0];
+        let serialized_transaction = serialize(&transaction).expect("Transaction should serailize");
+        let send_over_rpc = if let Some(tpu_client) = tpu_client {
+            !tpu_client
+                .send_wire_transaction(serialized_transaction.clone())
+                .await
+        } else {
+            true
+        };
+        if send_over_rpc {
+            if let Err(e) = rpc_client.send_transaction(&transaction).await {
+                match e.kind {
+                    solana_rpc_client_api::client_error::ErrorKind::Io(_)
+                    | solana_rpc_client_api::client_error::ErrorKind::Reqwest(_) => {
+                        // continue on io error, we will retry the transaction
```

Comment:
```suggestion
                        // fall through on io error, we will retry the transaction
```

so as not to be confused with `continue`